### PR TITLE
feat(app-metrics): Set up API for querying data

### PR DESCRIPTION
**Pull request by macobo**
## Problem

Depends on https://github.com/PostHog/posthog/pull/12143

Part of https://github.com/PostHog/posthog/issues/12009 / https://github.com/orgs/PostHog/projects/69/views/1

## Changes

3 new endpoints:
- `/api/projects/{id}/app_metrics/{plugin_config_id}` returns metric data (and totals) to be displayed on graphs for a given category.
- `/api/projects/{id}/app_metrics/{plugin_config_id}/historical_exports` lists historical exports that were run for this plugin
- `/api/projects/{id}/app_metrics/{plugin_config_id}/historical_exports/{job_id}` returns metric data and summary of a given export

## How did you test this code?

See unit tests

### CODE REVIEW

File: posthog/queries/app_metrics/app_metrics.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return relative_date_parse(self.filter.validated_data.get("date_from"))
+
+    @property
+    def date_to(self):
+        date_to_string = self.filter.validated_data.get("date_to")
+        return relative_date_parse(date_to_string) if date_to_string is not None else now()
+
+    @property
+    def interval(self) -> IntervalType:
+        if self.date_to - self.date_from < timedelta(days=2):
```
Comment (tiina303) [inline posthog/queries/app_metrics/app_metrics.py:67]: I'd let the api user decide this for their query

File: posthog/queries/app_metrics/serializers.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from rest_framework import serializers
+
+
+class AppMetricsRequestSerializer(serializers.Serializer):
+    category = serializers.ChoiceField(
+        # Keep in sync with plugin-server/src/worker/ingestion/app-metrics.ts
+        choices=["processEvent", "onEvent", "exportEvents"],
+        help_text="What date to filter the results from",
```
Comment (tiina303) [inline posthog/queries/app_metrics/serializers.py:8]: not date

File: posthog/queries/app_metrics/serializers.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        required=True,
+    )
+    date_from = serializers.CharField(
+        default="-30d",
+        help_text="What date to filter the results from. Can either be a date `2021-01-01`, or a relative date, like `-7d` for last seven days, `-1m` for last month, `mStart` for start of the month or `yStart` for the start of the year.",
+    )
+    date_to = serializers.CharField(
+        required=False,
+        help_text="What date to filter the results to. Can either be a date `2021-01-01`, or a relative date, like `-7d` for last seven days, `-1m` for last month, `mStart` for start of the month or `yStart` for the start of the year.",
```
Comment (tiina303) [inline posthog/queries/app_metrics/serializers.py:17]: could be helpful to know that it's exclusive range

File: posthog/queries/app_metrics/test/test_app_metrics.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        "team_id": team_id,
+        "plugin_config_id": plugin_config_id,
+        "category": category,
+        "job_id": job_id or "",
+        "successes": successes,
+        "successes_on_retry": successes_on_retry,
+        "failures": failures,
+    }
+    p = ClickhouseProducer()
+    p.produce(topic=KAFKA_APP_METRICS, sql=INSERT_APP_METRICS_SQL, data=data)
```
Comment (tiina303) [inline posthog/queries/app_metrics/test/test_app_metrics.py:37]: don't we need to wait for this to be ingested in tests to avoid flakiness?

**Review by tiina303 — APPROVED** [review]

File: posthog/queries/app_metrics/app_metrics.py (hunk 1 above)
Comment (macobo) [reply posthog/queries/app_metrics/app_metrics.py:67]: This is used potentially in two places:
1. Showing metrics on a single historical export
2. (Potentially) Allowing users to choose "past 24 hours" in a dropdown (we default to 30d).

In neither case does it make sense to make this an API param, especially for (1)

File: posthog/queries/app_metrics/serializers.py (hunk 3 above)
Comment (macobo) [reply posthog/queries/app_metrics/serializers.py:17]: This is copy-pasted from trends docs.

File: posthog/queries/app_metrics/test/test_app_metrics.py (hunk 4 above)
Comment (macobo) [reply posthog/queries/app_metrics/test/test_app_metrics.py:37]: No - test data is ingested instantly in python tests.

**Pull request merged**